### PR TITLE
gaphor: 2.8.2 -> 2.26.0

### PR DESCRIPTION
**Pull request by alex-nt**
## Description of changes

`gaphor` no longer builds, this will update it to `2.25.1`.

## Things done

* Added [better-exceptions](https://pypi.org/project/better-exceptions/) as it was the only missing package needed in latest `gaphor`
* Added all the new dependencies `gaphor` has

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by alex-nt** [comment]
Hi @WolfangAukang, can you please take a look at this? I was trying to migrate a system to nixos `24.05` and `gaphor` was no longer building. This is an attempt to make it work again.

**Comment by applePrincess** [comment]
Add explanation why test for `better-exceptions` is disabled
like: https://github.com/NixOS/nixpkgs/pull/283526/files#diff-94fc944559699062c299639eaa32e9f7130f30764619c6bdc2731d8fc0c7ac52R48-R50

**Comment by Clausinho** [comment]
LGTM? @WolfangAukang

**Review by JohnAZoidberg — CHANGES_REQUESTED** [review]
Doesn't build for me:

```
Checking runtime dependencies for gaphor-2.25.1-py3-none-any.whl
  - defusedxml<0.8.0,>=0.7.1 not satisfied by version 0.8.0rc2
```

I used `nixpkgs-review pr 316688` on AMD64

**Comment by Scrumplex** [comment]
https://github.com/gaphor/gaphor/releases/tag/2.26.0 has been released recently

**Comment by alex-nt** [comment]
@JohnAZoidberg  It seems to work now. @Scrumplex updated, thanks for letting me know.

**Comment by getchoo** [comment]
babel already has its own PR for staging in https://github.com/NixOS/nixpkgs/pull/334520. Would probably be better to wait on that

File: pkgs/development/python-modules/pycairo/default.nix [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 buildPythonPackage rec {
   pname = "pycairo";
-  version = "1.26.0";
+  version = "1.26.1";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pycairo/default.nix:17]: This doesn't make sense without changing the hash.

File: pkgs/tools/misc/gaphor/default.nix [hunk 2]
```diff
@@ -4,51 +4,66 @@
...
 , makeDesktopItem
 , python
 }:
 
 buildPythonApplication rec {
   pname = "gaphor";
-  version = "2.8.2";
-
+  version = "2.26.0";
   format = "pyproject";
```
Comment (dotlambda) [inline pkgs/tools/misc/gaphor/default.nix:32]:
```suggestion
  pyproject = true;
```

File: pkgs/tools/misc/gaphor/default.nix [hunk 3]
```diff
@@ -4,51 +4,66 @@
...
   ];
 
   buildInputs = [
-    gtksourceview4
+    gtksourceview5
     pango
+    libadwaita
   ];
 
   propagatedBuildInputs = [
```
Comment (dotlambda) [inline pkgs/tools/misc/gaphor/default.nix:54]:
```suggestion
  dependencies = [
```

File: pkgs/tools/misc/gaphor/default.nix
Comment (dotlambda) [inline pkgs/tools/misc/gaphor/default.nix:1]: `build-system` is missing

File: pkgs/development/python-modules/better-exceptions/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  # After disabling and running the build, no tests are collected.
+  doCheck = false;
+
+  src = fetchPypi {
+    pname = "better_exceptions";
+    inherit version;
+    hash = "sha256-5Oa8GERNXwTm6JSxA4Hl6SHT1UQkBBgWLH21fp6zRTs=";
+  };
+
+  meta = with lib; {
```
Comment (dotlambda) [inline pkgs/development/python-modules/better-exceptions/default.nix:22]:
```suggestion
  meta = {
```

File: pkgs/development/python-modules/better-exceptions/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    description = "Pretty and more helpful exceptions in Python, automatically.";
+    homepage = "https://github.com/qix-/better-exceptions";
+    license = licenses.mit;
```
Comment (dotlambda) [inline pkgs/development/python-modules/better-exceptions/default.nix:25]:
```suggestion
    license = lib.licenses.mit;
```

File: pkgs/development/python-modules/better-exceptions/default.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+}:
+
+buildPythonPackage rec {
+  pname = "better-exceptions";
+  version = "0.3.3";
+  format = "setuptools";
```
Comment (dotlambda) [inline pkgs/development/python-modules/better-exceptions/default.nix:10]:
```suggestion
  pyproject = true;
```
and add `build-system = [ setuptools ]`

File: pkgs/development/python-modules/better-exceptions/default.nix [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+  # As noted by @WolfangAukang, some check files need to be disabled because of various errors, same with some tests.
+  # After disabling and running the build, no tests are collected.
+  doCheck = false;
```
Comment (dotlambda) [inline pkgs/development/python-modules/better-exceptions/default.nix:14]: This is usually put below the `src` and all dependencies.

File: pkgs/development/python-modules/better-exceptions/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/better-exceptions/default.nix:1]: `meta.changelog` and `meta.maintainers` missing

**Review by dotlambda — COMMENTED** [review]
Please turn this into one commit per package.

File: pkgs/development/python-modules/better-exceptions/default.nix
Comment (alex-nt) [reply pkgs/development/python-modules/better-exceptions/default.nix:1]: `meta.changelog` is a bit weird for this version, well, for most versions after `0.2.1` I can't find something official.  

Who should I add for `meta.maintainers`? I know frightfully little about how everything works 😅.

Comment (dotlambda) [reply pkgs/development/python-modules/better-exceptions/default.nix:1]: > Who should I add for `meta.maintainers`?

Preferably yourself.

**Review by dotlambda — COMMENTED** [review]
Reminder: one commit per package

File: pkgs/development/python-modules/better-exceptions/default.nix [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    inherit version;
+    hash = "sha256-5Oa8GERNXwTm6JSxA4Hl6SHT1UQkBBgWLH21fp6zRTs=";
+  };
+
+  build-system = [ setuptools ];
+
+  # As noted by @WolfangAukang, some check files need to be disabled because of various errors, same with some tests.
+  # After disabling and running the build, no tests are collected.
+  doCheck = false;
+
```
Comment (natsukium) [inline pkgs/development/python-modules/better-exceptions/default.nix:24]: Could you add `pythonImportsCheck` for a smoke test?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck-using-pythonimportscheck

File: pkgs/development/python-modules/better-exceptions/default.nix [hunk 9]
```diff
@@ -26,5 +28,6 @@ buildPythonPackage rec {
     description = "Pretty and more helpful exceptions in Python, automatically.";
     homepage = "https://github.com/qix-/better-exceptions";
     license = lib.licenses.mit;
+    maintainers = lib.maintainers.alex-nt;
```
Comment (dotlambda) [inline pkgs/development/python-modules/better-exceptions/default.nix]:
```suggestion
    maintainers = [ lib.maintainers.alex-nt ];
```

**Comment by alex-nt** [comment]
> Reminder: one commit per package

Should I merge all commits into one?

**Comment by dotlambda** [comment]
> > Reminder: one commit per package
> 
> Should I merge all commits into one?

No, there should be three commits: one adding yourself to the maintainer list and one per package.

**Comment by alex-nt** [comment]
I'll move this to a new branch as I was stuck in `rebase` hell for way too much time today.

**Comment by alex-nt** [comment]
https://github.com/NixOS/nixpkgs/pull/335627/commits

**Pull request closed without merging**